### PR TITLE
channel argument for '/forcespawn'

### PR DESCRIPTION
**Pull request by eliaswolfy**
I don't know python, so go easy on my code, please?

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -3147,21 +3147,26 @@ async def fake(message: discord.Interaction):
...
         await message.response.send_message("bro what", ephemeral=True)
         return
-
+    channel_id = message.channel.id
     try:
-        if db["cat"][str(message.channel.id)]:
+        if channel:
+            channel_id = channel.id
+    except Exception:
+        pass
```
Comment (milena-kos) [inline main.py:3161]: what would be the Exception here?

File: main.py [hunk 2]
```diff
@@ -3147,21 +3147,26 @@ async def fake(message: discord.Interaction):
...
-@discord.app_commands.describe(cat_type="select a cat type ok")
+@discord.app_commands.describe(cat_type="select a cat type ok", channel="another channel then")
 @discord.app_commands.autocomplete(cat_type=cat_type_autocomplete)
-async def forcespawn(message: discord.Interaction, cat_type: Optional[str]):
+async def forcespawn(message: discord.Interaction, cat_type: Optional[str], channel: Optional[discord.TextChannel]):
     if cat_type and cat_type not in cattypes:
         await message.response.send_message("bro what", ephemeral=True)
         return
-
+    channel_id = message.channel.id
```
Comment (milena-kos) [inline main.py:3156]:
```suggestion
    if not channel:
        channel_id = message.channel.id
```

Comment (eliaswolfy) [reply main.py:3156]: I don't know how to handle optional arguments, I was afraid of an error happening

**Comment by milena-kos** [comment]
im just gonna edit the rest myself

File: main.py [hunk 3]
```diff
@@ -3147,21 +3147,26 @@ async def fake(message: discord.Interaction):
 @bot.tree.command(description="(ADMIN) Force cats to appear")
 @discord.app_commands.default_permissions(manage_guild=True)
 @discord.app_commands.rename(cat_type="type")
-@discord.app_commands.describe(cat_type="select a cat type ok")
+@discord.app_commands.describe(cat_type="select a cat type ok", channel="another channel then")
 @discord.app_commands.autocomplete(cat_type=cat_type_autocomplete)
-async def forcespawn(message: discord.Interaction, cat_type: Optional[str]):
+async def forcespawn(message: discord.Interaction, cat_type: Optional[str], channel: Optional[discord.TextChannel]):
```
Comment (milena-kos) [inline main.py:3152]: summon channels can also be VoiceChannel and Thread im pretty sure

Comment (eliaswolfy) [reply main.py:3152]: I didn't know, sorry

Comment (milena-kos) [reply main.py:3152]: im not blaming you, just showing things to fix :3

Comment (eliaswolfy) [reply main.py:3152]: I should have seen more about the bot

Comment (eliaswolfy) [reply main.py:3152]: ok, Github didn't load your message.
thanks, Milenakos!

**Pull request closed without merging**